examples/Microsoft.Spark.CSharp.Examples/Sql/VectorUdfs.cs

```diff
@@ -56,39 +58,27 @@ public void Run(string[] args)
             spark.Stop();
         }
 
-        private static RecordBatch CountCharacters(
-            RecordBatch records,
+        private static FxDataFrame CountCharacters(
+            FxDataFrame dataFrame,
             string groupFieldName,
             string stringFieldName)
         {
-            int stringFieldIndex = records.Schema.GetFieldIndex(stringFieldName);
-            StringArray stringValues = records.Column(stringFieldIndex) as StringArray;
-
             int characterCount = 0;
 
-            for (int i = 0; i < stringValues.Length; ++i)
+            PrimitiveColumn<int> characterCountColumn = new PrimitiveColumn<int>(stringFieldName + "CharCount");
+            PrimitiveColumn<int> ageColumn = new PrimitiveColumn<int>(groupFieldName);
+            for (long i = 0; i < dataFrame.RowCount; i++)
             {
-                string current = stringValues.GetString(i);
-                characterCount += current.Length;
+                characterCount += ((string)dataFrame[stringFieldName][i]).Length;
+            }
+            if (dataFrame.RowCount > 0)
+            {
+                characterCountColumn.Append(characterCount);
+                ageColumn.Append((int?)dataFrame[groupFieldName][0]);
             }
 
-            int groupFieldIndex = records.Schema.GetFieldIndex(groupFieldName);
-            Field groupField = records.Schema.GetFieldByIndex(groupFieldIndex);
-
-            // Return 1 record, if we were given any. 0, otherwise.
-            int returnLength = records.Length > 0 ? 1 : 0;
-
-            return new RecordBatch(
-                new Schema.Builder()
-                    .Field(groupField)
-                    .Field(f => f.Name(stringFieldName + "_CharCount").DataType(Int32Type.Default))
-                    .Build(),
-                new IArrowArray[]
-                {
-                    records.Column(groupFieldIndex),
```

Comment:
RecordBatch doesn't allow variable length ArrowArrays right? `records.Column(groupFieldIndex)` could have a Length of more than 1 here?

Reply:
RecordBatch allows it. It doesn't enforce that each array is of the same length. So you can create 2 columns:

1. Length 10
2. Length 5

And wrap a RecordBatch around them and say the RecordBatch's length is `5`. This allows you to reuse buffers/arrays without copying data just to truncate the length.

I'm not sure if this is great behavior or not, but it is how it works currently. I can look how the C++ and Java Arrow libraries work w.r.t. this.